core/src/banking_stage.rs

```diff
@@ -747,19 +725,27 @@ impl BankingStage {
     fn process_loop(
         packet_deserializer: &mut PacketDeserializer,
         poh_recorder: &Arc<RwLock<PohRecorder>>,
-        cluster_info: &ClusterInfo,
+        cluster_info: Arc<ClusterInfo>,
         recv_start: &mut Instant,
         id: u32,
         transaction_status_sender: Option<TransactionStatusSender>,
         replay_vote_sender: ReplayVoteSender,
-        data_budget: &DataBudget,
+        data_budget: Arc<DataBudget>,
         log_messages_bytes_limit: Option<usize>,
         connection_cache: Arc<ConnectionCache>,
         bank_forks: &Arc<RwLock<BankForks>>,
         mut unprocessed_transaction_storage: UnprocessedTransactionStorage,
     ) {
+        let forwarder = Forwarder::new(
```

Comment:
i think `.process_loop()` should rather take `forwarder` as an arg rather than creating here. so, this `*er` service `struct`s are lined up beautifully.

```rust

    fn process_loop(
        packet_deserializer: &mut PacketDeserializer,
        poh_recorder: &Arc<RwLock<PohRecorder>>,
        ...
        transaction_status_sender: Option<TransactionStatusSender>,
        ...
        forwarder: Forwarder
```

also, the name `process_loop` implies just _loop_ after any needed setup. that's another reason to move out.

Reply:
100% agree - I guess my brain was saving this for a separate PR but probably better to just do it here 😄

Reply:
Done: cd73736428029fc4bf0ae5ae02008d41bff59521